include/Clipboard.h

```diff
@@ -29,25 +29,40 @@
 #include <QDomElement>
 
 
-class JournallingObject;
-
 class Clipboard
 {
 public:
-	typedef QMap<QString, QDomElement> Map;
-
-	static void copy( JournallingObject * _object );
-	static const QDomElement * getContent( const QString & _node_name );
-
-	static const char * mimeType()
+	enum MimeType
```

Comment:
I thought about that, but wouldn't that result in a bit more verbose code when we need to access a MimeType? With `enum` we just need to type `Clipboard::StringPair` for example, while with `enum class` we would need `Clipboard::MimeType::StringPair` afaik. Unless we had the MimeType enum outside the class, but then there's the problem of using another name from the global namespace. I don't know, if you still think it's worth it to change it I can do it.

Reply:
It would indeed be more verbose, but that's not necessarily a bad thing, and `enum class`es have a couple of advantages over `enum`s.
Firstly, they're more strongly typed - they aren't implicitly convertible to `int` - and the C++ Core Guidelines recommend them over `enum`s primarily for this reason ([Enum.3](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Renum-class)).
Secondly, they offer better organisation and scoping of their enumerators. Plain `enum`s inject their enumerators, a group of closely related constants, into the containing scope, where they sit equally among less related things. This can make it hard to tell what's related to what without reading the header itself. (Does `Clipboard::Default` mean the default clipboard? The default key? `Clipboard::MimeType::Default` is unambiguous.) It can also lead to name collisions if another `enum` is defined in the future, e.g. `enum ClipboardType { Default, Selection, FindBuffer };`.

An `enum` works fine, so I won't require it to be changed, but I do prefer `enum class`.

Reply:
They are all valid and important points, I think the extra verbosity is worth all the benefits you mentioned. I just used `sed` to replace the enum values and should be committing later. On the process I might have found a small bug which I'm trying to find more about (a small misbehavior when cloning BB Tracks in an specific situation), but I should commit anyways since it is almost surely unrelated to the enum changes.